decode.go

```diff
@@ -47,7 +48,9 @@ func decodeFloat(t reflect.Type) decodeFunc {
 	bits := t.Bits()
 	return func(s string, v reflect.Value) error {
 		n, err := strconv.ParseFloat(s, bits)
-		if err != nil {
+		if err != nil && s == "" {
```

Comment:
Could it be like this?

```go
	return func(s string, v reflect.Value) error {
                if s == "" {
                  v.SetFloat(math.Nan())
                  return nil
                }

		n, err := strconv.ParseFloat(s, bits)
		if err != nil {
			return &UnmarshalTypeError{Value: s, Type: t}
		}
		v.SetFloat(n)
		return nil
	}
```

Reply:
I think either way is valid. From a performance standpoint. I'm not sure how costly the comparison is. But this would only be faster in the case where nulls have a higher frequency in the dataset than non null.

Reply:
it is valid, the code does the right thing, but it is idiomatic to keep the indent flow minimal for readability.

Reply:
that way works than. Also, just double checked and it looks like pandas defaults writing NaN floats to "" in `to_csv()` so I think this is the right default to make.

Reply:
Thank you for looking into that default value! We definitely should make it optional in the Encoder, perhaps it could work as a field like NaNFormat or something like this. I am not sure if we should make empty string a default because it could potentially change the behavior of someone's existing code, but let me think about it.